### PR TITLE
[nrf5-lock] start joiner role on boot

### PR DESCRIPTION
**Pull request by bukepo**
#### Problem
This commit starts joiner role boot boot so that it can be discovered by commissioners.

 #### Summary of Changes
Start joiner role within 15 minutes on boot.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp [hunk 1]
```diff
@@ -114,6 +115,24 @@ BaseType_t GenericThreadStackManagerImpl_FreeRTOS<ImplClass>::SignalThreadActivi
...
+template <class ImplClass>
+void GenericThreadStackManagerImpl_FreeRTOS<ImplClass>::OnJoinerTimer(TimerHandle_t xTimer)
+{
+    GenericThreadStackManagerImpl_FreeRTOS<ImplClass> * self =
+        static_cast<GenericThreadStackManagerImpl_FreeRTOS<ImplClass> *>(pvTimerGetTimerID(xTimer));
+
+    if (xTaskGetTickCount() > self->mJoinerExpire || self->Impl()->IsThreadProvisioned())
+    {
+        ChipLogDetail(DeviceLayer, "Thread joiner timer stopped.");
+        xTimerStop(xTimer, 0);
```
Comment (andy31415) [inline src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp:127]: should we check return codes in some form of ASSERT ?

File: src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp [hunk 2]
```diff
@@ -127,6 +146,12 @@ void GenericThreadStackManagerImpl_FreeRTOS<ImplClass>::ThreadTaskMain(void * ar
     // Capture the Thread task handle.
     self->mThreadTask = xTaskGetCurrentTaskHandle();
 
+    // Try starting joiner within 15m.
+    self->mJoinerExpire = xTaskGetTickCount() + pdMS_TO_TICKS(15 * 60 * 1000);
+
+    TimerHandle_t joinerTimer = xTimerCreate("JoinerTimer", pdMS_TO_TICKS(10000), pdTRUE, self, &OnJoinerTimer);
```
Comment (andy31415) [inline src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp:152]: check for NULL return in case of failure?

**Review by gjc13 — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

File: src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp (hunk 2 above)
Comment (woody-apple) [reply src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp:154]: @andy31415 ?

File: src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp (hunk 1 above)
Comment (woody-apple) [reply src/platform/FreeRTOS/GenericThreadStackManagerImpl_FreeRTOS.ipp]: @andy31415 ?

**Review by mspang — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**